keras_nlp/models/phi3/phi3_decoder.py

```diff
@@ -0,0 +1,262 @@
+
+        self.layer_norm_epsilon = layer_norm_epsilon
+        self.activation = keras.activations.get(activation)
+        self.kernel_initializer = keras.initializers.get(kernel_initializer)
+
+    def build(self, decoder_sequence_shape):
+
+        # Pre-attention layernorm.
+        self._pre_attention_layernorm = Phi3LayerNorm(
+            epsilon=self.layer_norm_epsilon,
+            dtype=self.dtype_policy,
+            name="pre_attention_layernorm",
+        )
+        self._pre_attention_layernorm.build(decoder_sequence_shape)
+
+        # Self attention layer.
+        self._attention = Phi3Attention(
+            num_query_heads=self.num_query_heads,
+            num_key_value_heads=self.num_key_value_heads,
+            kernel_initializer=clone_initializer(self.kernel_initializer),
+            dropout=self.dropout,
+            max_sequence_length=self.max_sequence_length,
+            original_max_sequence_length=self.original_max_sequence_length,
+            rope_max_wavelength=self.rope_max_wavelength,
+            rope_scaling_type=self.rope_scaling_type,
+            rope_scaling_short_factor=self.rope_scaling_short_factor,
+            rope_scaling_long_factor=self.rope_scaling_long_factor,
+            dtype=self.dtype_policy,
+            name="attention",
+        )
+        self._attention.build(decoder_sequence_shape)
+
+        # Post-attention layernorm.
+        self._post_attention_layernorm = Phi3LayerNorm(
+            epsilon=self.layer_norm_epsilon,
+            dtype=self.dtype_policy,
+            name="post_attention_layernorm",
+        )
+        self._post_attention_layernorm.build(decoder_sequence_shape)
+
+        # feedforward layers.
+        self._feedforward_intermediate_dense = keras.layers.Dense(
+            self.intermediate_dim,
+            kernel_initializer=clone_initializer(self.kernel_initializer),
+            use_bias=False,
+            dtype=self.dtype_policy,
+            name="feedforward_intermediate_dense",
+        )
+        self._feedforward_intermediate_dense.build(decoder_sequence_shape)
+
+        self._feedforward_gate_dense = keras.layers.Dense(
+            self.intermediate_dim,
+            kernel_initializer=clone_initializer(self.kernel_initializer),
+            use_bias=False,
+            dtype=self.dtype_policy,
+            name="feedforward_gate_dense",
+        )
+        self._feedforward_gate_dense.build(decoder_sequence_shape)
+
+        self._feedforward_output_dense = keras.layers.Dense(
+            self.hidden_dim,
+            kernel_initializer=clone_initializer(self.kernel_initializer),
+            use_bias=False,
+            dtype=self.dtype_policy,
+            name="feedforward_output_dense",
+        )
+
+        self._feedforward_output_dense.build(
+            self._feedforward_gate_dense.compute_output_shape(
+                decoder_sequence_shape
+            )
+        )
+
+        # Dropout
+        if self.dropout > 0.0:
```

Comment:
I think we usually just make the dropout layers and apply them regardless of value. (It will just pass through if dropout is 0)

Makes for simpler code.